src/unix/fs.c

```diff
@@ -157,6 +157,49 @@ extern char *mkdtemp(char *template); /* See issue #740 on AIX < 7 */
   while (0)
 
 
+/* Returns the positive fractional part of a floating-point number.
+ * Garbage in, garbage out: don't pass in NaN or Infinity.
+ *
+ * Open-coded (sort of, it's not exactly equivalent to modf(3))
+ * to avoid a dependency on libm.
+ */
+static double uv__frac(double x) {
+  if (x < 0)
+    x = -x;
+
+  x -= (uint64_t) x;
+
+  return x;
+}
+
+
+UV_UNUSED(static struct timespec uv__to_timespec(double x)) {
+  struct timespec t;
+
+  t.tv_sec = x;
+  t.tv_nsec = 1e9 * uv__frac(x);
+
+  /* TODO(bnoordhuis) Remove this. utimesat() has nanosecond resolution but we
```

Comment:
Out of curiosity, when can we remove this? I'm only asking because WASI wants nanosecond precision.

Reply:
I'll merge this and open a PR that removes it.